### DISCUSSION TITLE
European Fuel and Topography Support

### DISCUSSION THREAD

**Discussion by doyled-it** [mitrefireline/simfire#6] [Ideas]
## Overview & Problem

I've had a couple requests via email about adding support for European locations. Right now, this would require a pretty substantial change through separate fuel and topography APIs. **Both the fuel and topography** layers are currently downloaded for the United States through the [python-landfire package](https://pypi.org/project/landfire/) (a python wrapper around the [LANDFIRE](https://landfire.gov/) API), and as of the creation of this discussion, **a similar python API wrapper does not exist for a European fuel and topography dataset** (that I can find).

This means that integration into SimFire would require:
1. Identifying the correct fuel data source to download from.
2. Identifying the correct topographic data source to download from.
3. Creating a python wrapper to download the fuel and topography data for the requested area in Europe through an API.
4. Attaching the python wrapper to SimFire's fuel and topography data layers.

Unfortunately, the SimFire team doesn't have the current capacity to develop this functionality ourselves, but we can advise, review and edit Pull Requests, and inform on issues.

### Fuel

With some very light googling, I've found a possible European fuel source with a simple URL request API. It would just require writing a Python wrapper:

- [EFFIS](https://effis.jrc.ec.europa.eu/about-effis/technical-background/fuels)

### Topography

I have no idea if these would work in practice, but some options:

- [EUDEM](https://www.opentopodata.org/datasets/eudem/)
  - You'd have to build a python wrapper around this API.
- [PyGMT](https://www.pygmt.org/latest/index.html)
  - I'm not sure if the data that PyGMT gives us is what SimFire requires for topography.

## Possible Solutions

1. (Recommended) Create a python package similar to `python-landfire` and incorporate into SimFire as a dependency, deciding to use this package instead of `python-landfire`, dependent on lat/long.
    - I see this being generally useful for more than just this project, so I recommend this approach more than `2.`. 
3. Include python wrapper for external API inside of SimFire (messier than and not as extensible as `1.`).

**Comment by robinnarsinghranabhat** [comment]
Hi. Found this in https://community.climatechange.ai post section. 
I would love to contribute. How should I proceed ? 

To Add : 
I understand the high level idea to implement this. But I am unfamiliar with Landfire API. Am currently checking out some of their videos in their  youtube channel.

**Reply by doyled-it** [reply]
I think the best way to go about it is to first create a python wrapper around the EFFIS fuel data API, so that a user can request a fuel rectangle based on lat/long corners.

Similar to [this usage](https://landfire-python.firesci.io/en/latest/landfire.html).
